### PR TITLE
IM101

### PR DESCRIPTION
**Pull request by BEAdi**
## Related Issues
fixes: https://jira-dc.paloaltonetworks.com/browse/CIAC-9799

## Description
Validate refactor- IM101 check that image is not too large

### CODE REVIEW

File: demisto_sdk/commands/validate/validators/IM_validators/IM101_image_too_large.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    description = "Checks that the image file dimensions are matching the requirements."
+    rationale = "Image needs to fit its place in the UI. For more information see: https://xsoar.pan.dev/docs/integrations/integration-logo"
+    error_message = "You've created/modified a yml or package with a large sized image. Please make sure to change the image dimensions at:"
+    related_field = "image"
+    is_auto_fixable = False
+    expected_git_statuses = [GitStatuses.ADDED, GitStatuses.MODIFIED]
+    related_file_type = [RelatedFileType.IMAGE]
+
+    def is_valid(self, content_items: Iterable[ContentTypes]) -> List[ValidationResult]:
+        image_max_size = 10 * 1024  # 10kB
```
Comment (jbabazadeh) [inline demisto_sdk/commands/validate/validators/IM_validators/IM101_image_too_large.py:27]: maybe we want to add it as a const?

File: demisto_sdk/commands/validate/tests/IM_validators_test.py [hunk 2]
```diff
@@ -41,6 +44,34 @@ def test_ImageExistsValidator_is_valid_image_path():
...
+
+    When:
+        - Calling the ImageTooLargeValidator is_valid function.
+
+    Then:
+        - Make sure the right amount of integration image path failed, and that the right error message is returned.
+        - Case 1: Should fail.
+        - Case 2: Shouldn't fail.
+    """
+    content_items = [
```
Comment (jbabazadeh) [inline demisto_sdk/commands/validate/tests/IM_validators_test.py:62]: Please change the unit tests to use pytest's parametrize feature for clarification which
test should pass/fail

**Review by jbabazadeh — COMMENTED** [review]
Great job!
Please see my comments.

File: demisto_sdk/commands/validate/validators/IM_validators/IM101_image_too_large.py [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+IMAGE_MAX_SIZE = 10 * 1024  # 10kB
+
+ContentTypes = Integration
+
+
+class ImageTooLargeValidator(BaseValidator[ContentTypes]):
+    error_code = "IM101"
+    description = "Checks that the image file dimensions are matching the requirements."
+    rationale = "Image needs to fit its place in the UI. For more information see: https://xsoar.pan.dev/docs/integrations/integration-logo"
+    error_message = "You've created/modified a yml or package with a large sized image. Please make sure to change the image dimensions at:"
```
Comment (YuvHayun) [inline demisto_sdk/commands/validate/validators/IM_validators/IM101_image_too_large.py:22]:
```suggestion
    error_message = "You've created/modified a yml or package with a large sized image. Please make sure to change the image dimensions at: {0}."
```

File: demisto_sdk/commands/validate/validators/IM_validators/IM101_image_too_large.py [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    related_field = "image"
+    is_auto_fixable = False
+    expected_git_statuses = [GitStatuses.ADDED, GitStatuses.MODIFIED]
```
Comment (YuvHayun) [inline demisto_sdk/commands/validate/validators/IM_validators/IM101_image_too_large.py:25]:
```suggestion
```
Remove both lines.

**Review by YuvHayun — APPROVED** [review]
Great job!
minor comments.

**Review by jbabazadeh — APPROVED** [review]
Nice!

**Pull request merged**